### PR TITLE
[BTS-1856][3.11] Query: Protect access on member _execStats

### PR DESCRIPTION
**Pull request by hkernbach**
### Scope & Purpose

This PR resolves a data-race during parallel accesses on the `_execStats` class variable.
See TSAN report here: https://arangodb.atlassian.net/browse/BTS-1856

- [x] :hankey: Bugfix

### Checklist

- [x] Tests: No new tests added. Found by nightly TSAN run.
- [x] :book: CHANGELOG entry made
- [x] Backports (Yes this fix should be added to all supported versions)
  - [x] Backport for 3.12.0: https://github.com/arangodb/arangodb/pull/20851
  - [x] Backport for 3.11: This is 3.11 :) 
  - [ ] Backport for 3.10: No, EOL (As of April 15, 2024, support for ArangoDB 3.10 will end.)

#### Related Information

- [x] GitHub issue / Jira ticket: https://arangodb.atlassian.net/browse/BTS-1856

### CODE REVIEW

**Comment by hkernbach** [comment]
TSAN Run: https://jenkins.arangodb.biz/job/arangodb-ANY-linux-san.x86-64/2394/

File: arangod/Aql/Query.cpp [hunk 1]
```diff
@@ -1617,7 +1618,7 @@ futures::Future<Result> finishDBServerParts(Query& query, ErrorCode errorCode) {
         network::sendRequest(pool, "server:" + server, fuerte::RestVerb::Delete,
                              "/_api/aql/finish/" + std::to_string(queryId),
                              body, options)
-            .thenValue([ss, &query](network::Response&& res) mutable -> Result {
+            .thenValue([&query](network::Response&& res) mutable -> Result {
```
Comment (jsteemann) [inline arangod/Aql/Query.cpp:1621]: Is this modification safe?
Now we are only capturing query by reference, but we do not ensure that the query is still valid when the `then` is executed. Previously we have captured a shared-ptr copy of the query state `ss`, and in its `executeLocked()` method also checked if the state was still valid.
This is now all gone, and I fear the change is not safe.

**Review by jsteemann — CHANGES_REQUESTED** [review]
See https://github.com/arangodb/arangodb/pull/20849/files#r1567684193

File: arangod/Aql/Query.cpp (hunk 1 above)
Comment (hkernbach) [reply arangod/Aql/Query.cpp:1621]: Yes, correct. Will fix.

**Comment by hkernbach** [comment]
https://jenkins.arangodb.biz/job/arangodb-ANY-linux-san.x86-64/

**Review by jsteemann — APPROVED** [review]
LGTM

**Comment by hkernbach** [comment]
https://jenkins.arangodb.biz/job/arangodb-ANY-linux-san.x86-64/2396/

**Review by mchacki — APPROVED** [review]
LGTM!

**Pull request merged**